components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.regex/src/main/java/org/wso2/carbon/identity/login/resolver/regex/RegexLoginResolver.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.login.resolver.regex;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.login.resolver.mgt.LoginResolver;
+import org.wso2.carbon.identity.login.resolver.mgt.ResolvedUserResult;
+import org.wso2.carbon.identity.login.resolver.regex.utils.UserResolverUtil;
+import org.wso2.carbon.user.api.Claim;
+import org.wso2.carbon.user.api.ClaimManager;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UniqueIDUserStoreManager;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.common.AuthenticationResult;
+import org.wso2.carbon.user.core.common.User;
+import org.wso2.carbon.user.core.constants.UserCoreClaimConstants;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * This class defines the default implementation of the LoginResolver. With the help of this login resolver, users will
+ * get resolved using a claim regex pattern defined in the management console.
+ */
+public class RegexLoginResolver implements LoginResolver {
+
+    private static final Log log = LogFactory.getLog(RegexLoginResolver.class);
+
+    /**
+     * Resolves a user from the given login identifier and then returns the resolved claim URI and the user details if
+     * a matching user exists.
+     *
+     * @param loginIdentifier   The user login identifier input which is provided by the user during login.
+     * @param allowedAttributes The list of claim URIs which are allowed by admin for resolving the user.
+     * @param tenantDomain      The tenant domain of the user.
+     * @return A ResolvedUserResult object with the user details and the user resolved claim details.
+     */
+    @Override
+    public ResolvedUserResult resolveUser(String loginIdentifier, List<String> allowedAttributes, String tenantDomain) {
```

Comment:
Would be better if we could add debug level logs as well just in case if we need to debug the code. WDYT ?